### PR TITLE
Feature/healthcheck

### PR DESCRIPTION
**Pull request by gedge**
### What

add healthcheck checking of downstream services (mongo, neo, dataset-api)

### How to review

qv https://github.com/ONSdigital/go-ns/pull/55

### CODE REVIEW

File: cmd/dp-filter-api/main.go [hunk 1]
```diff
@@ -99,6 +108,8 @@ func main() {
 
 		cancel()
 
+		healthTicker.Close()
```
Comment (CarlHembrough) [inline cmd/dp-filter-api/main.go]: Consider calling close on the ticker before closing the mongo session.

Comment (gedge) [reply cmd/dp-filter-api/main.go]: moved

File: dataset/api.go [hunk 2]
```diff
@@ -171,6 +172,11 @@ func (api *API) callDatasetAPI(ctx context.Context, method, path string, payload
 	return jsonBody, resp.StatusCode, nil
 }
 
+// GetHealthCheckClient returns a healthcheck-compatible client
+func (api *API) GetHealthCheckClient() *datasetclient.Client {
+	return datasetclient.New(api.url)
```
Comment (CarlHembrough) [inline dataset/api.go]: Consider inlining `datasetclient.New(api.url)` in main when the ticker is created.

Comment (gedge) [reply dataset/api.go]: As discussed, I was trying to have all `datasetAPI` access in one package, and allow for a possible future merging of this client and the `go-ns` client.  Seemed cleaner to me, so hope we can agree to differ.

**Pull request merged**